src/rpc/rpc_reward_api.py

```diff
@@ -376,8 +380,13 @@ def do_rpc_request(self, request, time_out=120):
                 "RPC URL '{}' not found. Is this node in archive mode?".format(request)
             )
 
-        # URL returned something broken
-        if resp.status_code != HTTPStatus.OK:
+        # URL returned something broken from the client side 4xx
+        # server side errors 5xx can pass for a retry
+        if (
+            HTTPStatus.BAD_REQUEST
+            <= resp.status_code
+            < HTTPStatus.INTERNAL_SERVER_ERROR
+        ):
```

Comment:
Relevant code section regarding [this issue](https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/issues/511). 
If the response status code is greater or equal to BAD_REQUEST (400) and less then INTERNAL_SERVER_ERROR (500) the raise the APIProvider Exception which won't trigger a retry in the while loop.

Reply:
Anything 400-499 is an error we shouldn't return, I would say anything bellow 499

Reply:
I think it makes sense to raise the exception just for client side errors which are 400-499 everything else should be able to return so that the while loop continues.

Are you suggesting that a status code 3 can exists and we should cover that edge case?
![image](https://user-images.githubusercontent.com/14346967/149674965-1f696bf7-c6a8-4eea-b65d-278635c15048.png)

Reply:
Any chances of getting 3xx, 1xx? If almost none, than your code is fine. I agree 5xx should retry.